.github/workflows/test.yaml

```diff
@@ -38,8 +38,18 @@ jobs:
         - functionality/memory
         - functionality/planner
         - functionality/hooks
+        - test_controller
         
     steps:
       - uses: actions/checkout@v4
       - uses: astral-sh/setup-uv@v5
-      - run: uv run --with=dotenv pytest tests/${{ matrix.test }}.py || true
+      - name: Setup virtual environment and install dependencies
+        if: matrix.test == 'test_controller'
+        run: |
+          uv venv
+          uv pip install -e .
+          uv pip install patchright
```

Comment:
```suggestion
```
I believe none of is is needed, `astral-sh/setup-uv@v5` already sets up the venv with all the packges needed